plugins/modules/purefa_vg.py

```diff
@@ -224,148 +236,187 @@ def get_multi_vgroups(module, destroyed=False):
+                if LooseVersion(CONTEXT_API_VERSION) <= LooseVersion(api_version):
+                    res = array.create_volume_groups(
+                        context_names=[module.params["context"]],
+                        names=[module.params["name"]],
+                        volume_group=VolumeGroupPost(
+                            qos=Qos(iops_limit=module.params["iops_qos"])
+                        ),
+                    )
+                else:
+                    res = array.create_volume_groups(
+                        names=[module.params["name"]],
+                        volume_group=VolumeGroupPost(
+                            qos=Qos(iops_limit=module.params["iops_qos"])
+                        ),
+                    )
+                if res.status_code != 200:
+                    module.fail_json(
+                        msg="Vgroup {0} creation failed. Error: {1}".format(
+                            module.params["name"], res.errors[0].message
                         )
-            else:
-                module.fail_json(
-                    msg="IOPs QoS value {0} out of range.".format(
-                        module.params["iops_qos"]
                     )
-                )
         else:
-            bw_qos_size = int(human_to_bytes(module.params["bw_qos"]))
-            if int(human_to_real(module.params["iops_qos"])) in range(
-                100, 100000000
-            ) and bw_qos_size in range(1048576, 549755813888):
-                changed = True
-                if not module.check_mode:
-                    try:
-                        array.create_vgroup(
-                            module.params["name"],
-                            iops_limit=module.params["iops_qos"],
-                            bandwidth_limit=module.params["bw_qos"],
-                        )
-                    except Exception:
-                        module.fail_json(
-                            msg="Vgroup {0} creation failed.".format(
-                                module.params["name"]
+            module.fail_json(
+                msg="IOPs QoS value {0} out of range.".format(module.params["iops_qos"])
+            )
+    else:
+        bw_qos_size = int(human_to_bytes(module.params["bw_qos"]))
+        if int(human_to_real(module.params["iops_qos"])) in range(
+            100, 100000000
+        ) and bw_qos_size in range(1048576, 549755813888):
+            changed = True
+            if not module.check_mode:
+                if LooseVersion(CONTEXT_API_VERSION) <= LooseVersion(api_version):
+                    res = array.create_volume_groups(
+                        context_names=[module.params["context"]],
+                        names=[module.params["name"]],
+                        volume_group=VolumeGroupPost(
+                            qos=Qos(
+                                iops_limit=module.params["iops_qos"],
+                                bandwidth_limit=module.params["bw_qos"],
```

Comment:
directly passing qos params causing error, should be using iops_qos_size & bw_qos_size